pyproject.toml

```diff
@@ -155,6 +155,8 @@ black = "~22.8.0"
 coverage = "~6.4.1"
 # Tool for debugging Django
 django-debug-toolbar = "~3.7.0"
+# DiscoverSlowestTestsRunner for running CI performance tests and benchmarking.
+django-slowtests = "^1.1.1"
```

Comment:
In general I'm OK with this being a dev dependency, **but** we need to think about the case of plugins using our test runner to run their own tests. (similar to #2536). Plugins shouldn't have to know about Nautobot's dev-only dependencies in order to run their own unit tests; it should be possible to run `nautobot-server test <plugin>` (or `coverage run --module nautobot.core.cli test <plugin>`, equivalently) with only Nautobot's core dependencies plus the plugin's own dev dependencies installed.

I'm thinking for #2536 I'll probably address this with a settings flag like `TEST_USE_FACTORIES` and appropriate conditional logic around imports and calls to ensure that the factory logic is only called when opted into (i.e. always by core, conditionally by plugins if and only if they decide to opt into and support this feature).

Reply:
Just to clarify, this conditional logic will leave `factory-boy` and `django-slowtests` as dev dependencies and only when the plugin decide to opt in to have those test features using a settings flag, they will become available as a plugin own dev dependencies? 
Can we make them as plugin dev dependencies to begin with or is it not worth it?

Reply:
If the plugin wants to use these test features, they will have to add those packages as their own dev dependencies. Plugins have no direct visibility into Nautobot's own dev dependencies as those are a Poetry feature, not a Python package/wheel feature.

We could, I suppose, add these as optional primary dependencies under an extra (so you could do `pip install nautobot[test-infra]`) rather than as dev dependencies, which would at least make it easier for plugins to get a consistent version for these libraries... the more I think about it the more I like that idea. But even if we go that route, plugins still need to be able to continue to run their existing tests as-is; these add-ons still need to be opt-in.

What I'm trying to avoid is putting plugins in a position of "Your tests are going to blow up and refuse to run under Nautobot 1.5 unless/until you update your project."

Reply:
Thinking about it more, the `extras` approach probably isn't feasible, as IIRC Poetry doesn't let you have different extras dependencies for the same package under different sections, i.e. a plugin *could not* do:

```toml
[tool.poetry.dependencies]
nautobot = "*"

[tool.poetry.dev-dependencies]
nautobot = {"version": "*", extras=["test-infra"]}
```

See #2570 for the approach I ended up going with for factory-boy.